guides/FFI.md

```diff
@@ -156,45 +156,11 @@ This time, we can assign the curried function type directly:
 foreign import calculateInterest :: Number -> Number -> Number
 ```
 
-#### Handling Constrained Types
+#### Don't call functions with constrained types from JavaScript
 
-Another special case that you should be aware of when calling PureScript functions from Javascript is that values with constrained types (i.e. types which contain type class constraints) contain extra parameters which are used to pass type class dictionaries to the function.
+It's not safe to call functions that have typeclass constraints from JavaScript, as their representation is unstable in the face of compiler optimizations. If you want to call a PureScript function from JavaScript you need to take all arguments explicitly.
```

Comment:
I think you could clarify this a little. All arguments on the JS side are explicit, but the point is that some arguments would have been filled in by the compiler. I think you could clarify that you can export a function for use in JS with a monomorphic signature which will result in the compiler pre-applying the dictionary.

Reply:
Thanks for the feedback, I struggled with expressing exactly what you're saying here. I'll sit down and try to formulate it.